packages/viz/__tests__/SizeBinsStyle.spec.ts

```diff
@@ -0,0 +1,166 @@
+      const sizeBinStyleInstance = sizeBinsStyle('attributeName', {
+        bins: [0, 10, 15, 20],
+        sizes: [5, 15, 30]
+      });
+
+      expect(sizeBinStyleInstance.getRadius(recordExample)).toBe(15);
+      expect(sizeBinStyleInstance.getRadius(recordOutOfRangeExample)).toBe(
+        defaultSizeBinsOptions.othersSize
+      );
+      expect(sizeBinStyleInstance.getRadius(recordNullValueExample)).toBe(
+        defaultSizeBinsOptions.nullSize
+      );
+      expect(sizeBinStyleInstance.getLineWidth(recordExample)).toBe(15);
+      expect(sizeBinStyleInstance.getLineWidth(recordOutOfRangeExample)).toBe(
+        defaultSizeBinsOptions.othersSize
+      );
+      expect(sizeBinStyleInstance.getLineWidth(recordNullValueExample)).toBe(
+        defaultSizeBinsOptions.nullSize
+      );
+    });
+    it('should returns the radius and line width for each feature', () => {
+      const sizeBinStyleInstance = sizeBinsStyle('attributeName', {
+        bins: [0, 10, 15, 20],
+        sizes: [5, 15, 30],
+        othersSize: 60,
+        nullSize: 100
+      });
+
+      expect(sizeBinStyleInstance.getRadius(recordExample)).toBe(15);
+      expect(sizeBinStyleInstance.getRadius(recordOutOfRangeExample)).toBe(60);
+      expect(sizeBinStyleInstance.getRadius(recordNullValueExample)).toBe(100);
+      expect(sizeBinStyleInstance.getLineWidth(recordExample)).toBe(15);
+      expect(sizeBinStyleInstance.getLineWidth(recordOutOfRangeExample)).toBe(
+        60
+      );
+      expect(sizeBinStyleInstance.getLineWidth(recordNullValueExample)).toBe(
+        100
+      );
+    });
+  });
+
+  describe('min/max values', () => {
+    it('should gets the min/max values from bins', () => {
+      const sizeBinStyleInstance = sizeBinsStyle('attributeName', {
+        bins: [0, 10, 15, 20],
+        sizes: [5, 15, 30]
+      });
+
+      expect(sizeBinStyleInstance.pointRadiusMinPixels).toBe(
+        Math.min(
+          5,
+          defaultSizeBinsOptions.othersSize,
+          defaultSizeBinsOptions.nullSize
+        )
+      );
+      expect(sizeBinStyleInstance.pointRadiusMaxPixels).toBe(
+        Math.max(
+          30,
+          defaultSizeBinsOptions.othersSize,
+          defaultSizeBinsOptions.nullSize
+        )
+      );
+      expect(sizeBinStyleInstance.lineWidthMinPixels).toBe(
+        Math.min(
+          5,
+          defaultSizeBinsOptions.othersSize,
+          defaultSizeBinsOptions.nullSize
+        )
+      );
+      expect(sizeBinStyleInstance.lineWidthMaxPixels).toBe(
+        Math.max(
+          30,
+          defaultSizeBinsOptions.othersSize,
+          defaultSizeBinsOptions.nullSize
+        )
+      );
+    });
+    it('should gets the min/max values from all parameters', () => {
+      const sizeBinStyleInstance = sizeBinsStyle('attributeName', {
+        bins: [0, 10, 15, 20],
+        sizes: [5, 15, 30],
+        othersSize: 1,
+        nullSize: 60
```

Comment:
Have you tried using nullSize: 0 ? Does it work? Maybe it could be a first way to filter out unwanted values

Reply:
good point! I added a test to cover that case also. Yes, by this way we can filter out null values with `nullSize` or features out of the range with `otherSize`